cmd/snapd-generator/main.c

```diff
@@ -185,7 +185,7 @@ int ensure_fusesquashfs_inside_container(const char *normal_dir)
 			fprintf(stderr, "cannot open %s: %m\n", fname);
 			return 2;
 		}
-		fprintf(f, "[Mount]\nType=%s\n", fstype);
+		fprintf(f, "[Mount]\nType=%s\nOptions=nodev,ro,x-gdu.hide,allow_other\nLazyUnmount=yes\n", fstype);
```

Comment:
Do we have a link to the code before this was a generator and what options we set back then?

Reply:
It is spread over systemd.go (see mountUnitTemplate and fsMountOptions) and fstype.go in a bit convoluted way; I've copied them from mount options of a manually installed snap inside the container.

Reply:
this will be applied to all types, yet i thought in gocode the "allow_other" at least is scoped to fuse mount types only.

I somehow expected something like `if strcmp() ==0 || strcmp == 0` to `fprintf(",allow_other)"` or some such (pseudo code).

Or it is safe to always pass allow_other?

It would be nice if the test would be somehow interactive.... can we compare the generated units in `/run/systemd/generator` with the ones in `/etc/systemd/system` ? Such that if and when mount options are changed anywhere in the gocode, the generator unittest fails due to any differences?

I guess as long as unit tests do run under lxd and non-lxd at least sometimes?

Reply:
@xnox this part of the generator is run for squashfuse/snapfuse only (fstype is either "fuse.squashfuse" or "fuse.snapfuse"). Both support "allow_other".

The idea for the test is nice, I'll see what I can do here or in a followup.

Reply:
Yeah, found it in the code

Reply:
So the extra test check (comparing Options of snapd-generated unit & snapd-generated unit) is possible and I'm working on it but fighting some issues around these checks, I think I'll leave it for a followup tomorrow, don't want to block this PR.

Reply:
Done in a followup #9617